### PR TITLE
Cleanup: split core.views into more manageable modules

### PR DESCRIPTION
**Pull request by julen**
Most notably decorators and mixins have been put apart. For the latter the
import has been adjusted to `core.views.mixins` to explicitly warn about mixin
usage.

### CODE REVIEW

**Comment by phlax** [comment]
looks good

> has been adjusted to core.views.mixins to explicitly warn about mixin
usage.

i didnt see the warning - maybe i didnt check properly.

**Comment by julen** [comment]
There is no warning as in a code warning such as `warnings.warn` or similar but a hint by forcing to use `.mixins` in the import. Sorry if it wasn't clear enough; I'll adjust the commit message.

File: pootle/apps/reports/views.py [hunk 1]
```diff
@@ -20,19 +20,18 @@
...
                               JsonResponseNotFound)
 from pootle.core.log import PAID_TASK_ADDED, PAID_TASK_DELETED, log
 from pootle.core.utils.timezone import make_aware, make_naive
-from pootle.core.views import AjaxResponseMixin, UserObjectMixin
+from pootle.core.views.mixins import (AjaxResponseMixin, NoDefaultUserMixin,
+                                      TestUserFieldMixin, UserObjectMixin)
 from pootle_misc.util import (ajax_required, get_date_interval,
                               get_max_month_datetime, import_func)
-from pootle_profile.views import (DetailView, NoDefaultUserMixin,
-                                  TestUserFieldMixin)
```
Comment (unho) [inline pootle/apps/reports/views.py]: Nice one!

**Comment by unho** [comment]
lgtm once issues with import sorting are sorted.

**Pull request merged**